src/module-elasticsuite-core/Model/System/Message/WarningAboutClusterShardsMisconfig.php

```diff
@@ -0,0 +1,169 @@
+<?php
+/**
+ * DISCLAIMER
+ * Do not edit or add to this file if you wish to upgrade Smile Elastic Suite to newer
+ * versions in the future.
+ *
+ * @category  Smile
+ * @package   Smile\ElasticsuiteCore
+ * @author    Romain Ruaud <romain.ruaud@smile.fr>
+ * @copyright 2022 Smile
+ * @license   Open Software License ("OSL") v. 3.0
+ */
+
+namespace Smile\ElasticsuiteCore\Model\System\Message;
+
+use Magento\Framework\Notification\MessageInterface;
+use Magento\Framework\UrlInterface;
+use Smile\ElasticsuiteCore\Helper\IndexSettings as IndexSettingsHelper;
+use Smile\ElasticsuiteIndices\Model\IndexStatsProvider;
+
+/**
+ * ElasticSuite Warning about Cluster mis-configuration for shards
+ *
+ * @category Smile
+ * @package  Smile\ElasticsuiteCore
+ * @author   Vadym Honcharuk <vahonc@smile.fr>
+ *
+ * @SuppressWarnings(PHPMD.LongVariable)
+ */
+class WarningAboutClusterShardsMisconfig implements MessageInterface
+{
+    /**
+     * Route to Stores -> Configuration section
+     */
+    private const ROUTE_SYSTEM_CONFIG = 'adminhtml/system_config/edit';
+
+    /**
+     * Anchor for Stores -> Configuration -> ELASTICSUITE -> Base Settings -> Indices Settings
+     */
+    private const ANCHOR_ES_INDICES_SETTINGS_PATH = 'smile_elasticsuite_core_base_settings_indices_settings-link';
+
+    /**
+     * URL for Elasticsuite Indices Settings wiki page
+     */
+    private const ES_INDICES_SETTINGS_WIKI_PAGE = 'https://github.com/Smile-SA/elasticsuite/wiki/ModuleInstall#indices-settings';
+
+    /**
+     * @var IndexSettingsHelper
+     */
+    protected $helper;
+
+    /**
+     * @var IndexStatsProvider
+     */
+    protected $indexStatsProvider;
+
+    /**
+     * @var UrlInterface
+     */
+    private $urlBuilder;
+
+    /**
+     * @param IndexSettingsHelper $indexSettingHelper Index settings helper
+     * @param IndexStatsProvider  $indexStatsProvider Index stats provider
+     * @param UrlInterface        $urlBuilder         Url builder
+     */
+    public function __construct(
+        IndexSettingsHelper    $indexSettingHelper,
+        IndexStatsProvider $indexStatsProvider,
+        UrlInterface $urlBuilder
+    ) {
+        $this->helper             = $indexSettingHelper;
+        $this->indexStatsProvider = $indexStatsProvider;
+        $this->urlBuilder         = $urlBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isDisplayed()
+    {
+        $numberOfShards = $this->helper->getNumberOfShards();
+        $indexMaxSize = $this->getElasticsuiteIndexMaxSize()['size_in_bytes'];
```

Comment:
You could move this inside the ```if ($numberOfShards > 1)```, by doing so, we would only retrieve the indices collection when it's needed, so this will be more performant.

Reply:
```['size_in_bytes']``` could be undefined if there are no indices actually. You should add some guard functions to check that ```getElasticsuiteIndexMaxSize()``` is returning something, and if the result has this key.

Reply:
Fixed.